### PR TITLE
Helm: Add flags to toggle metamonitoring logs/metrics custom resources

### LINKED ISSUE #2937: [HELM] -  Add the ability to disable any one of meta monitoring instances
#### Is your feature request related to a problem? Please describe.
Whenever we enable meta monitoring, both logs and metrics are enabled by default. There is no way to disable one of them.

#### Describe the solution you'd like
Provide a flag to help disable any one component.

### PR DESCRIPTION
**Pull request by shubhamranjan**
#### What this PR does

Adds the capability to disable any single meta monitoring component.

#### Which issue(s) this PR fixes or relates to

Fixes #2937 
Fixes https://github.com/grafana/mimir/issues/2892

#### Checklist

- [x] Tests updated
- [x] Documentation added
- [x] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

File: operations/helm/charts/mimir-distributed/values.yaml [hunk 1]
```diff
@@ -1882,6 +1882,9 @@ metaMonitoring:
     installOperator: false
 
     logs:
+      # Controls wherther to create resources related to PodLogs and LogsInstance
```
Comment (dimitarvdimitrov) [inline operations/helm/charts/mimir-distributed/values.yaml:1885]:
```suggestion
      # -- Controls whether to create resources PodLogs and LogsInstance resources
```

File: operations/helm/charts/mimir-distributed/values.yaml [hunk 2]
```diff
@@ -1904,6 +1907,9 @@ metaMonitoring:
       additionalClientConfigs: []
 
     metrics:
+      # Controls wherther to create resources related to MetricsInstance and ServiceMonitor
```
Comment (dimitarvdimitrov) [inline operations/helm/charts/mimir-distributed/values.yaml:1910]:
```suggestion
      # -- Controls whether to create MetricsInstance resources and ServiceMonitor resources for scraping Kubernetes (when .scrapeK8s.enabled=true).
```

**Review by dimitarvdimitrov — APPROVED** [review]
Thank you! I left two comments with wording suggestions.

**Review by dimitarvdimitrov — APPROVED** [review]
LGTM! Thank you! You need to rebase onto grafana:main before I can merge this

**Comment by pracucci** [comment]
@dimitarvdimitrov or @krajorama could you take a final look, please?

File: operations/helm/charts/mimir-distributed/values.yaml (hunk 1 above)
Comment (dimitarvdimitrov) [reply operations/helm/charts/mimir-distributed/values.yaml:1885]: I don't think this is resolved

File: operations/helm/charts/mimir-distributed/values.yaml (hunk 2 above)
Comment (dimitarvdimitrov) [reply operations/helm/charts/mimir-distributed/values.yaml:1910]: I don't think this is resolved

File: CHANGELOG.md [hunk 3]
```diff
@@ -22,6 +22,7 @@
 * [ENHANCEMENT] Add sanity check at startup to ensure the configured filesystem directories don't overlap for different components. #2828
 * [ENHANCEMENT] Go: updated to go 1.19.1. #2637
 * [ENHANCEMENT] Runtime config: don't unmarshal runtime configuration files if they haven't changed. This can save a bit of CPU and memory on every component using runtime config. #2954
+* [ENHANCEMENT] Monitoring: Add additional flags to conditionally enable log / metric scraping.
```
Comment (dimitarvdimitrov) [inline CHANGELOG.md:25]:
```suggestion
* [ENHANCEMENT] Monitoring: Add additional flags to conditionally enable log / metric scraping. #2936
```

**Review by dimitarvdimitrov — CHANGES_REQUESTED** [review]

**Comment by shubhamranjan** [comment]
My Apologies ! I had somehow messed the commits before. Have fixed now and corrected changes are back,

File: .github/CODEOWNERS [hunk 4]
```diff
@@ -1 +0,0 @@
-* @mimir-maintainers
```
Comment (dimitarvdimitrov) [inline .github/CODEOWNERS:1]: I'm not sure what happened here

Comment (dimitarvdimitrov) [reply .github/CODEOWNERS:1]: I also see a lot of other changes. you can try starting clean from grafana:main onto a branch in your fork, cherry-pick aa62e3cad104e142ba019d71fa67cafa9d256225 3a37b08db679f11a1e66aee84285261dfa9ede21 and 20de41934fc7e752ead4bfb0a6a7129f7a435aaf and force push

Comment (shubhamranjan) [reply .github/CODEOWNERS:1]: yeah, really messed up this one.

File: CHANGELOG.md [hunk 5]
```diff
@@ -23,6 +23,7 @@
 * [ENHANCEMENT] Add sanity check at startup to ensure the configured filesystem directories don't overlap for different components. #2828
 * [ENHANCEMENT] Go: updated to go 1.19.1. #2637
 * [ENHANCEMENT] Runtime config: don't unmarshal runtime configuration files if they haven't changed. This can save a bit of CPU and memory on every component using runtime config. #2954
+* [ENHANCEMENT] Monitoring: Add additional flags to conditionally enable log / metric scraping. #2936
```
Comment (krajorama) [inline CHANGELOG.md:26]: Please move the changelog entry to the helm chart changelog in `operations/helm/charts/mimir-distributed/CHANGELOG.md`.  See for reference: https://github.com/grafana/mimir/blob/main/docs/internal/contributing/contributing-to-helm-chart.md

File: operations/helm/charts/mimir-distributed/templates/metamonitoring/pod-logs.yaml [hunk 6]
```diff
@@ -1,5 +1,5 @@
 {{- with (.Values.metaMonitoring).grafanaAgent }}
-{{- if .enabled }}
+{{- if and (.enabled) (.logs.enabled) }}
```
Comment (krajorama) [inline operations/helm/charts/mimir-distributed/templates/metamonitoring/pod-logs.yaml:2]:
```suggestion
{{- if and .enabled .logs.enabled }}
```

File: operations/helm/charts/mimir-distributed/templates/metamonitoring/metrics-instance.yaml [hunk 7]
```diff
@@ -1,5 +1,5 @@
 {{- with .Values.metaMonitoring.grafanaAgent }}
-{{- if .enabled }}
+{{- if and (.enabled) (.metrics.enabled) }}
```
Comment (krajorama) [inline operations/helm/charts/mimir-distributed/templates/metamonitoring/metrics-instance.yaml:2]:
```suggestion
{{- if and .enabled .metrics.enabled }}
```

File: operations/helm/charts/mimir-distributed/templates/metamonitoring/metrics-instance-usernames-secret.yaml [hunk 8]
```diff
@@ -1,5 +1,5 @@
-{{- if .Values.metaMonitoring.grafanaAgent.enabled }}
-{{- with .Values.metaMonitoring.grafanaAgent.metrics }}
+{{- with .Values.metaMonitoring.grafanaAgent }}
+{{- if and (.enabled) (.metrics.enabled) }}
```
Comment (krajorama) [inline operations/helm/charts/mimir-distributed/templates/metamonitoring/metrics-instance-usernames-secret.yaml:2]:
```suggestion
{{- if and .enabled .metrics.enabled }}
```

File: operations/helm/charts/mimir-distributed/templates/metamonitoring/logs-instance.yaml [hunk 9]
```diff
@@ -1,5 +1,5 @@
 {{- with (.Values.metaMonitoring).grafanaAgent }}
-{{- if .enabled }}
+{{- if and (.enabled) (.logs.enabled) }}
```
Comment (krajorama) [inline operations/helm/charts/mimir-distributed/templates/metamonitoring/logs-instance.yaml:2]:
```suggestion
{{- if and .enabled .logs.enabled }}
```

File: operations/helm/charts/mimir-distributed/templates/metamonitoring/logs-instance-usernames-secret.yaml [hunk 10]
```diff
@@ -1,4 +1,4 @@
-{{- if ((.Values.metaMonitoring).grafanaAgent).enabled }}
+{{- if and (.Values.metaMonitoring.grafanaAgent.enabled) (.Values.metaMonitoring.grafanaAgent.logs.enabled) }}
```
Comment (krajorama) [inline operations/helm/charts/mimir-distributed/templates/metamonitoring/logs-instance-usernames-secret.yaml:1]:
```suggestion
{{- if and .Values.metaMonitoring.grafanaAgent.enabled .Values.metaMonitoring.grafanaAgent.logs.enabled }}
```

File: operations/helm/charts/mimir-distributed/templates/metamonitoring/kubelet-cadvisor-servmon.yaml [hunk 11]
```diff
@@ -1,4 +1,4 @@
-{{- if and ((.Values.metaMonitoring).grafanaAgent).enabled ((((.Values.metaMonitoring).grafanaAgent).metrics).scrapeK8s).enabled }}
+{{- if and (.Values.metaMonitoring.grafanaAgent.enabled) (.Values.metaMonitoring.grafanaAgent.metrics.enabled) (.Values.metaMonitoring.grafanaAgent.metrics.scrapeK8s.enabled) }}
```
Comment (krajorama) [inline operations/helm/charts/mimir-distributed/templates/metamonitoring/kubelet-cadvisor-servmon.yaml:1]:
```suggestion
{{- if and .Values.metaMonitoring.grafanaAgent.enabled .Values.metaMonitoring.grafanaAgent.metrics.enabled .Values.metaMonitoring.grafanaAgent.metrics.scrapeK8s.enabled }}
```

File: operations/helm/charts/mimir-distributed/templates/metamonitoring/kube-state-metrics-servmon.yaml [hunk 12]
```diff
@@ -1,4 +1,4 @@
-{{- if and ((.Values.metaMonitoring).grafanaAgent).enabled ((((.Values.metaMonitoring).grafanaAgent).metrics).scrapeK8s).enabled }}
+{{- if and (.Values.metaMonitoring.grafanaAgent.enabled) (.Values.metaMonitoring.grafanaAgent.metrics.enabled) (.Values.metaMonitoring.grafanaAgent.metrics.scrapeK8s.enabled) }}
```
Comment (krajorama) [inline operations/helm/charts/mimir-distributed/templates/metamonitoring/kube-state-metrics-servmon.yaml:1]:
```suggestion
{{- if and .Values.metaMonitoring.grafanaAgent.enabled .Values.metaMonitoring.grafanaAgent.metrics.enabled .Values.metaMonitoring.grafanaAgent.metrics.scrapeK8s.enabled }}
```

**Review by krajorama — CHANGES_REQUESTED** [review]
Thank you for the contribution. I've made some comments and suggest removing the remaining parenthesis as well while we're at it

**Comment by shubhamranjan** [comment]
> Thank you for the contribution. I've made some comments and suggest removing the remaining parenthesis as well while we're at it

Alright. Sure about the parenthesis  thing ? I left them as is  because they are part of logical and operator and wanted to show segregation of conditions.

Just confirming, I can adhere to whatever style guide you would prefer.

**Comment by dimitarvdimitrov** [comment]
previously I included the explicit parenthesis so that users modifying the default values.yaml file from the chart wouldn't break rendering, but I think that's unnecessary. I'm fine with @krajorama's suggestions

**Review by dimitarvdimitrov — APPROVED** [review]
LGTM, provided Krajo also signs this off

**Comment by shubhamranjan** [comment]
@dimitarvdimitrov - Please review again, for some reason the review request button for you removes the review request for @krajorama

**Review by krajorama — APPROVED** [review]

**Pull request merged**